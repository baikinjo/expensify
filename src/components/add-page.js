/* Imports ======================================================================================= */
import React from 'react'
import { connect } from 'react-redux'

import Form from './form'
import { startAddExpense } from '../actions/expenses'

export class AddPage extends React.Component {
  onSubmit = expense => {
    this.props.startAddExpense(expense)
    this.props.history.push('/')
  }

  render() {
    return (
      <div>
        <h1>Add Expense</h1>
        <Form onSubmit={this.onSubmit} />
      </div>
    )
  }
}

const mapDispatchToProps = dispatch => ({
  startAddExpense: expense => dispatch(startAddExpense(expense))
})

export default connect(
  undefined,
  mapDispatchToProps
)(AddPage)
