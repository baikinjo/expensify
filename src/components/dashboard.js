/* Imports ======================================================================================= */
import React from 'react'

import List from './list'
import ListFilters from './list-filters'

const Dashboard = () => (
  <div>
    <ListFilters />
    <List />
  </div>
)

export default Dashboard
