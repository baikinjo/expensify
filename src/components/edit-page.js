/* Imports ======================================================================================= */
import React from 'react'

const EditPage = props => {
  console.log(props)
  return <div>Edit Expense</div>
}

export default EditPage
