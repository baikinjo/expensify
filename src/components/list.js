import React from 'react'
import { connect } from 'react-redux'

import ListItem from './list-item'
import selectExpenses from '../selectors/expenses'

export const List = props => (
  <div>
    {props.expenses.length === 0 ? (
      <p>No Expenses</p>
    ) : (
      props.expenses.map(expense => {
        return <ListItem key={expense.id} {...expense} />
      })
    )}
  </div>
)

const mapStateToProps = state => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  }
}

export default connect(mapStateToProps)(List)
