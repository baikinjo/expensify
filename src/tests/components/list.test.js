import React from 'react'
import { shallow } from 'enzyme'

import { List } from '../../components/list'
import expenses from '../fixtures/expenses'

test('should render List with expenses', () => {
  const wrapper = shallow(<List expenses={expenses} />)
  expect(wrapper).toMatchSnapshot()
})

test('should render List with empty message', () => {
  const wrapper = shallow(<List expenses={[]} />)
  expect(wrapper).toMatchSnapshot()
})
