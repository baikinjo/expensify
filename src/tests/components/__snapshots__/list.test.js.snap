// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render List with empty message 1`] = `
<div>
  <p>
    No Expenses
  </p>
</div>
`;

exports[`should render List with expenses 1`] = `
<div>
  <ListItem
    amount={1203}
    createdAt={0}
    description="gum"
    id="1"
    key="1"
    note=""
  />
  <ListItem
    amount={10000000}
    createdAt={-345600000}
    description="rent"
    id="2"
    key="2"
    note=""
  />
  <ListItem
    amount={32}
    createdAt={345600000}
    description="credit"
    id="3"
    key="3"
    note=""
  />
</div>
`;
